src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -584,10 +569,12 @@ private struct CompilationInfo
                 public Compilation Compilation { get; }
                 public bool HasCompleteReferences { get; }
 
-                public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
+                public CompilationInfo(Compilation compilation, ProjectInfo projectInfo, bool hasCompleteReferences)
                 {
                     this.Compilation = compilation;
-                    this.HasCompleteReferences = hasCompleteReferences;
+
+                    // to have complete reference, projectInfo should be complete and we could create complete reference for this compilation.
+                    this.HasCompleteReferences = projectInfo.IsComplete && hasCompleteReferences;
```

Comment:
in my mind, ProjectInfo.IsComplete is whether we have all project info or not. and HasCompleteReference is whether this project has all references or not. and if projectInfo can't guarantee its information, HasCompleteReference shouldn't as well since IsComplete is higher requirement than HasCompleteReference.